plugins/modules/katello_sync_plan.py

```diff
@@ -80,85 +80,70 @@
 
 
 def main():
-    module = ForemanEntityAnsibleModule(
-        argument_spec=dict(
+    module = KatelloEntityApypieAnsibleModule(
+        entity_spec=dict(
             name=dict(required=True),
             description=dict(),
-            organization=dict(required=True),
             interval=dict(choices=['hourly', 'daily', 'weekly', 'custom cron'], required=True),
             enabled=dict(type='bool', required=True),
             sync_date=dict(required=True),
             cron_expression=dict(),
+        ),
+        argument_spec=dict(
             products=dict(type='list'),
         ),
         required_if=[
             ['interval', 'custom cron', ['cron_expression']],
         ],
     )
 
-    (entity_dict, state) = module.parse_params()
+    entity_dict = module.clean_params()
 
-    if entity_dict['interval'] != 'custom cron':
-        if 'cron_expression' in entity_dict:
-            module.fail_json(msg='"cron_expression" cannot be combined with "interval"!="custom cron".')
+    if (entity_dict['interval'] != 'custom cron') and ('cron_expression' in entity_dict):
+        module.fail_json(msg='"cron_expression" cannot be combined with "interval"!="custom cron".')
 
     module.connect()
 
-    entity_dict['organization'] = find_organization(module, entity_dict['organization'])
-    products = entity_dict.pop('products', None)
-
-    sync_plan = find_sync_plan(module, entity_dict['name'], entity_dict['organization'], failsafe=True)
+    entity_dict['organization'] = module.find_resource_by_name('organizations', name=entity_dict['organization'], thin=True)
+    scope = {'organization_id': entity_dict['organization']['id']}
+    entity = module.find_resource_by_name('sync_plans', name=entity_dict['name'], params=scope, failsafe=True)
 
-    entity_dict = sanitize_entity_dict(entity_dict, name_map)
+    products = entity_dict.pop('products', None)
 
-    changed, sync_plan = naildown_entity(SyncPlan, entity_dict, sync_plan, state, module)
+    changed, sync_plan = module.ensure_entity('sync_plans', entity_dict, entity, params=scope)
 
-    if products is not None:
-        products = find_products(module, products, entity_dict['organization'])
-        desired_product_ids = set(p.id for p in products)
-        current_product_ids = set(p.id for p in sync_plan.product)
+    if not module.desired_absent and products is not None:
+        products = module.find_resources_by_name('products', products, params=scope, thin=True)
+        desired_product_ids = set(p['id'] for p in products)
+        current_product_ids = set(p['id'] for p in entity['products']) if entity else set()
 
         if desired_product_ids != current_product_ids:
             if not module.check_mode:
                 product_ids_to_add = desired_product_ids - current_product_ids
                 if product_ids_to_add:
-                    sync_plan.add_products(data={'product_ids': list(product_ids_to_add)})
+                    payload = scope.copy()
```

Comment:
I thought you prefer `payload.update(scope)`? ;-)

Reply:
This must have been a day older.